nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryQuery.java

```diff
@@ -3848,4 +3850,100 @@ public void testQueryDefaultValueInComplex() {
         assertEquals(1, dml.size());
     }
 
+    @Test
+    public void testIsRecord() {
+        String queryRecords = "SELECT * FROM MyDocType WHERE ecm:isRecord = 1";
+        String queryNotRecord = "SELECT * FROM MyDocType WHERE ecm:isRecord = 0";
+
+        // no record present initially
+        DocumentModelList dml ;
+        dml = session.query(queryRecords);
```

Comment:
```java
DocumentModelList dml = session.query(queryRecords);
```

Reply:
I write it this way so that the following code more symmetrical and is easier to read; there are repeated blocks of query / check results. I'll move the variable declaration a bit higher.